src/puppetlabs/puppetdb/query/reports.clj

```diff
@@ -102,8 +117,10 @@
   (let [first-row (first report-rows)
         resource-events (->> report-rows
                              (reduce collapse-resource-events []))]
-    (assoc (select-keys first-row report-columns)
-      :resource_events resource-events)))
+    (-> (select-keys first-row report-columns)
+        (update-in [:metrics] (scf-utils/parse-db-json-fn))
+        (update-in [:logs] (scf-utils/parse-db-json-fn))
```

Comment:
You could save a line here with `(partial maptrans {[:metrics :logs] parse-db-json-fn})`